_docs/en/2.5.0/compile-by-docker.md

```diff
@@ -0,0 +1,73 @@
+---
+permalink: 2.5.0/docs/build/compile-by-docker/
+redirect_from:
+  - 2.5.0/docs/build/compile-by-docker/
+version: 2.5.0
+---
+
+## Download the docker image
+
+Pegasus encapsulates the building environments into [docker images](https://hub.docker.com/r/apache/pegasus/tags?page=1&name=env), you can build directly based on these environments.
+
+For example, you can use the image based on `Ubuntu 20.04`:
+```bash
+docker pull apache/pegasus:build-env-ubuntu2004
+```
+
+## Compilation
+
+Please refer to [Downloads](/docs/downloads) to fetch the sources under a directory (`/your/local/apache-pegasus-source`). Then run the following command:
+
+If you want to run tests, you should build Pegasus by the following command:
+
+```bash
+docker run -v /your/local/apache-pegasus-source:/root/pegasus \
+           apache/pegasus:build-env-ubuntu2004 \
```

Comment:
Is there an image specified for 2.5.0? Because the images might have been updated(remove some packages), it may cause build failures.

Reply:
Good advice! I think we could modify [build-push-env-docker.yml](https://github.com/apache/incubator-pegasus/blob/master/.github/workflows/build-push-env-docker.yml) to support build-env images to different branches, such as master, v2.5.0, etc.